shell/common/gin_converters/file_dialog_converter_gin_adapter.h

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2019 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#ifndef SHELL_COMMON_GIN_CONVERTERS_FILE_DIALOG_CONVERTER_GIN_ADAPTER_H_
+#define SHELL_COMMON_GIN_CONVERTERS_FILE_DIALOG_CONVERTER_GIN_ADAPTER_H_
+
+#include "gin/converter.h"
+#include "shell/common/native_mate_converters/file_dialog_converter.h"
+
+// TODO(deermichel): replace adapter with real implementation after removing
+// mate
+// -- this adapter forwards all conversions to the existing mate converter --
+// (other direction might be preferred, but this is safer for now :D)
+
+namespace gin {
+
+template <>
+struct Converter<file_dialog::DialogSettings> {
+  static v8::Local<v8::Value> ToV8(v8::Isolate* isolate,
+                                   const file_dialog::DialogSettings& in) {
+    return mate::ConvertToV8(isolate, in);
```

Comment:
I realize this has been merged now but can a follow up PR be raised to reverse this logic.  The conversion logic should be moved to the gin converter and the mate converter should just call `gin::ConvertToV8`.

The reasoning behind this is that eventually when we have converted everything to gin we will only have to remove a bunch of placeholder converters and not have to do the risky "copy paste" for every converter at the same time, we should be moving the converters slowly in PRs like this.

cc @deepak1556

Reply:
Yes i'm actually surprised that this got merged so fast. My plan is to move all the converters to gin (have placeholders for mate) in some PR's.. adding this one to my backlog as well 👍